src/client/GameWindow.cpp

```diff
@@ -577,6 +544,47 @@ void GameWindow::setUpText(
     card.body->setPosition(xBodyPosition, yBodyPosition);
 }
 
+bool GameWindow::onHexagonClick(sf::Vector2i clickPosition)
+{
+    bool isClickable = false;
+    std::array<int, 2> hexagonOnClick = {0, 0};    
+    int minimumDistance = WINDOW_LENGTH;
+
+    for (auto &mapTexture : mapTextureToDisplay)
+    {
+        for (unsigned j = 0; j < mapTexture.getSize(); j++)
+        {
+            if (gameEnginePtr->intersectPointRect(clickPosition, mapTexture.getSprite(j).getGlobalBounds()))
```

Comment:
Try 
```
if ( ! gameEnginePtr->intersectPointRect(clickPosition, mapTexture.getSprite(j).getGlobalBounds())){
    continue;
}
```
If it works it will remove one more level of indentation